distributed/client.py

```diff
@@ -1053,6 +1053,23 @@ def _update_scheduler_info(self):
         except EnvironmentError:
             logger.debug("Not able to query scheduler for identity")
 
+    @gen.coroutine
+    def _wait_for_workers(self, func, n):
+        while n and (yield func()) < n:
+            yield gen.sleep(0.1)
+
+    def wait_for_workers(self, n_workers=0):
+        """
+        Note: If you add more arguments, you will need to return a list of
+        futures, or assert that only 1 is called at a time.
+        """
+        if n_workers:
+            @gen.coroutine
+            def f():
+                info = yield self.scheduler.identity()
+                raise gen.Return(len(info['workers']))
+            return self.sync(self._wait_for_workers, f, n_workers)
```

Comment:
Rather than have the nested coroutine here I recommend that you unpack the definition of f in `_wait_for_workers`.  I think that this should do it.

```python
def _wait_for_workers(self, n_workers=0):
    info = yield self.scheduler.identity()
    while len(info['workers']) < n_workers:
        yield gen.sleep(0.1)
        info = yield self.scheduler.identity()

def wait_for_workers(self, n_workers=0):
    return self.sync(self._wait_for_workers, n_workers=n_workers)
```